presto-spi/src/test/java/com/facebook/presto/spi/TestPage.java

```diff
@@ -64,24 +64,42 @@ public void testGetRegionFromNoColumnPage()
     public void testCompactDictionaryBlocks()
             throws Exception
     {
-        Slice[] expectedValues = createExpectedValues(10);
-        BlockBuilder blockBuilder = BIGINT.createBlockBuilder(new BlockBuilderStatus(), expectedValues.length);
-        for (Slice expectedValue : expectedValues) {
-            BIGINT.writeLong(blockBuilder, expectedValue.length());
-        }
-        Block lengthsDictionary = blockBuilder.build();
+        int positionCount = 100;
 
         // Create 2 dictionary blocks with the same source id
         DictionaryId commonSourceId = randomDictionaryId();
-        DictionaryBlock commonSourceIdBlock1 = createDictionaryBlock(expectedValues, 100, commonSourceId);
-        DictionaryBlock commonSourceIdBlock2 = new DictionaryBlock(commonSourceIdBlock1.getPositionCount(), lengthsDictionary, commonSourceIdBlock1.getIds(), commonSourceId);
+        int commonDictionaryUsedPositions = 20;
+        int[] commonDictionaryIds = getDictionaryIds(positionCount, commonDictionaryUsedPositions);
+
+        // first dictionary contains "varbinary" values
+        Slice[] dictionaryValues1 = createExpectedValues(50);
+        SliceArrayBlock dictionary1 = new SliceArrayBlock(dictionaryValues1.length, dictionaryValues1);
+        DictionaryBlock commonSourceIdBlock1 = new DictionaryBlock(positionCount, dictionary1, commonDictionaryIds, commonSourceId);
+
+        // second dictionary block is "length(firstColumn)"
+        BlockBuilder dictionary2 = BIGINT.createBlockBuilder(new BlockBuilderStatus(), dictionary1.getPositionCount());
+        for (Slice expectedValue : dictionaryValues1) {
+            BIGINT.writeLong(dictionary2, expectedValue.length());
+        }
+        DictionaryBlock commonSourceIdBlock2 = new DictionaryBlock(positionCount, dictionary2.build(), commonDictionaryIds, commonSourceId);
 
-        // Create block with a different source id
-        DictionaryBlock randomSourceIdBlock = createDictionaryBlock(expectedValues, 100, randomDictionaryId());
+        // Create block with a different source id, dictionary size, used
+        int otherDictionaryUsedPositions = 30;
+        int[] otherDictionaryIds = getDictionaryIds(positionCount, otherDictionaryUsedPositions);
+        SliceArrayBlock dictionary3 = new SliceArrayBlock(70, createExpectedValues(70));
+        DictionaryBlock randomSourceIdBlock = new DictionaryBlock(positionCount, dictionary3, otherDictionaryIds);
 
         Page page = new Page(commonSourceIdBlock1, randomSourceIdBlock, commonSourceIdBlock2);
```

Comment:
This re-ordering is confusing. I didn't notice it and the entire test didn't make sense for me.